app/models.py

```diff
@@ -722,7 +733,7 @@ class ServiceTableMixin(object):
     # Service publishing time.
     service_id = db.Column(db.String, index=True, unique=True, nullable=False)
 
-    data = db.Column(JSON)
+    data = db.Column(JSON, nullable=False, default=dict)
```

Comment:
Why the default? It feels to me that the service should never be an empty dictionary, so this is just hiding a potential error.